### DISCUSSION TITLE
How to configure rbac for okta multiple roles

### DISCUSSION THREAD

**Discussion by bharathiraja1986** [provectus/kafka-ui#4270] [Q&A]
Hi Team,

Can you help me to configure okta multiple roles for kafkaui? Currently single role configuration in roles.yaml file is working fine. But if i add 2 roles then it is giving null error as below,

**Configuration:**

rbac:
  roles:
    - name: "readonly"
      clusters:
        - Mycluster
      subjects:
        - provider: oauth
          type: role
          value: "MyKafkaui"
      permissions:
        - resource: clusterconfig
          actions: [ "view" ]
        - resource: topic
          value: ".*"
          actions:
            - VIEW
            - MESSAGES_READ
        - resource: consumer
          value: ".*"
          actions: [ view ]
        - resource: schema
          value: ".*"
          actions: [ view ]
        - resource: connect
          value: ".*"
          actions: [ view ]
    - name: "AdminRole"
      clusters:
        - Mycluster
      subjects:
        - provider: oauth
          type: role
          value: "mykafkaAdmin"
      permissions:
        - resource: applicationconfig
          actions: all
        - resource: clusterconfig
          actions: all
        - resource: topic
          value: ".*"
          actions: all
        - resource: consumer
          value: ".*"
          actions: all

**Error:**
[kafka-ui] | 2023-10-03 00:43:24,261 ERROR [reactor-http-epoll-1] o.s.b.a.w.r.e.AbstractErrorWebExceptionHandler: [350e38db-22]  500 Server Error for HTTP GET "/api/authorization"
[kafka-ui] | java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because the return value of "com.provectus.kafka.ui.model.rbac.Permission.getActions()" is null
[kafka-ui] |    at com.provectus.kafka.ui.controller.AccessController.lambda$mapPermissions$4(AccessController.java:69)
[kafka-ui] |    Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException:

**Comment by bharathiraja1986** [comment]
Any help here ????

**Reply by timchenko-a** [reply]
Actions should be a list, while in `AdminRole` you define them as a strings.
Proper way will be to define them like:
```
actions: [ all ]
```

**Comment by mohammed-fouzan** [comment]
Can you please share your config files?

**Comment by germanosin** [comment]
Me and @Haarolean no longer support this repo.
Don't you mind to propose your request to our repo https://github.com/kafbat/kafka-ui
